modes/regression.py

```diff
@@ -25,7 +25,11 @@ def __init__(self, args, module_dir, tmp_dir):
 
         super(RegressionMode, self).__init__(args, module_dir, tmp_dir)
 
-        # TODO: argument validation logic
+        if args.prefs is not None:
```

Comment:
Let's collect all argument validation logic in .setup(). Object init should never fail.